doc/specs/KeyboardManager.md

```diff
@@ -0,0 +1,263 @@
+  - \~70% of users requested shortcuts, either app-level or OS level, with OS shortcuts being the most demanded. This helped us prioritize the different features we planned in the functional spec.
+
+- *Question*: If remapping your keyboard required a restart, would you still use this PowerToy?   
+  - Before design, we were intimately aware that this could be a pain point for many users. Over 60% of respondents answered 'Yes', providing confidence that a restart, if necessary, is possible.
+
+## 1.10 Goals / Non-Goals
+
+#### *Goals*
+- The KSM can be used to configure keypresses on Windows 10 with any built-in / external keyboard
+- The KSM provides an intuitive and lightweight UI experience (53% of survey respondents who use 3<sup>rd</sup> party tools wanted integration into Windows and/or a UI)
+- The KSM will be able to remap keys
+- The KSM will be able to edit OS-level shortcuts
+- The KSM will work with all keyboards, regardless of locale (Example: Japanese keyboard)
+
+#### *Stretch Goals*
+- The KSM will be able to edit app-level shortcuts
+
+#### *Non-Goals*
+- Require a reboot only when absolutely necessary (42% of survey respondents would not use this PowerToy if reboot was required)
+
+## 2. Requirements
+### 2.1 Functional Requirements Overview
+
+| No. | Requirement                                                                                                                                                                                          | Pri |
+| --- |----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --- |
+|  1  | Remap any button on keyboard to any other button on keyboard                                                                                                                                         | P0  |
+|  2  | OS-Level key shortcuts. Example: Making WinKey + G lock the computer.                                                                                                                                | P1  | 
+|  3  | App level keyboard shortcuts. Example: Going to the address bar in Chrome on a MacOS is CMD+L, on a PC, that would map to Alt-L.                                                                     | P2  |
+|  4  | Settings page that easily configures all supported scenarios.                                                                                                                                        | P1  |
+|  5  | The Powertoys Shortcut Guide needs to be aware of keyboard remaps. Example: If Winkey + G locks the computer the Shortcut Guide needs to list this as the method to lock a computer, not Winkey + L. | P1  |
+|  6  | A Mac and Linux ready profile that is prepopulated with commonly used remaps and adjustments.                                                                                                        | P1  |
```

Comment:
I would break these into separate items.  Explicitly calling out Mac and Linux as i bet they will be different work items